engine/src/main/java/org/pentaho/di/trans/steps/terafast/TeraFast.java

```diff
@@ -0,0 +1,475 @@
+
+  private Process process;
+
+  private OutputStream fastload;
+
+  private OutputStream dataFile;
+
+  private PrintStream dataFilePrintStream;
+
+  private List<Integer> columnSortOrder;
+
+  private RowMetaInterface tableRowMeta;
+
+  private SimpleDateFormat simpleDateFormat;
+
+  /**
+   * Constructor.
+   *
+   * @param stepMeta
+   *          the stepMeta.
+   * @param stepDataInterface
+   *          the stepDataInterface.
+   * @param copyNr
+   *          the copyNr.
+   * @param transMeta
+   *          the transMeta.
+   * @param trans
+   *          the trans.
+   */
+  public TeraFast( final StepMeta stepMeta, final StepDataInterface stepDataInterface, final int copyNr,
+    final TransMeta transMeta, final Trans trans ) {
+    super( stepMeta, stepDataInterface, copyNr, transMeta, trans );
+  }
+
+  /**
+   * Create the command line for a fastload process depending on the meta information supplied.
+   *
+   * @return The string to execute.
+   *
+   * @throws KettleException
+   *           Upon any exception
+   */
+  public String createCommandLine() throws KettleException {
+    if ( StringUtils.isBlank( this.meta.getFastloadPath().getValue() ) ) {
+      throw new KettleException( "Fastload path not set" );
+    }
+    final StringBuilder builder = new StringBuilder();
+    try {
+      final FileObject fileObject =
+        KettleVFS.getFileObject( environmentSubstitute( this.meta.getFastloadPath().getValue() ) );
+      final String fastloadExec = KettleVFS.getFilename( fileObject );
+      builder.append( fastloadExec );
+    } catch ( Exception e ) {
+      throw new KettleException( "Error retrieving fastload application string", e );
+    }
+    // Add log error log, if set.
+    if ( StringUtils.isNotBlank( this.meta.getLogFile().getValue() ) ) {
+      try {
+        FileObject fileObject =
+          KettleVFS.getFileObject( environmentSubstitute( this.meta.getLogFile().getValue() ) );
+        builder.append( " -e " );
+        builder.append( "\"" + KettleVFS.getFilename( fileObject ) + "\"" );
+      } catch ( Exception e ) {
+        throw new KettleException( "Error retrieving logfile string", e );
+      }
+    }
+    return builder.toString();
+  }
+
+  protected void verifyDatabaseConnection() throws KettleException {
+    // Confirming Database Connection is defined.
+    if ( this.meta.getDbMeta() == null ) {
+      throw new KettleException( BaseMessages.getString( PKG, "TeraFastDialog.GetSQL.NoConnectionDefined" ) );
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.pentaho.di.trans.step.BaseStep#init(org.pentaho.di.trans.step.StepMetaInterface,
+   *      org.pentaho.di.trans.step.StepDataInterface)
+   */
+  @Override
+  public boolean init( final StepMetaInterface smi, final StepDataInterface sdi ) {
+    this.meta = (TeraFastMeta) smi;
+    // this.data = (GenericStepData) sdi;
+    simpleDateFormat = new SimpleDateFormat( FastloadControlBuilder.DEFAULT_DATE_FORMAT );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.